internal/schemautil/userconfig/util.go

```diff
@@ -174,3 +174,11 @@ func DecodeKey(k string) string {
 func IsKeyTyped(k string) bool {
 	return regexp.MustCompile(`^.*_(boolean|integer|number|string|array|object)$`).MatchString(k)
 }
+
+func UnmarshalUserConfig(src interface{}) map[string]interface{} {
+	configList, ok := src.([]interface{})
+	if !ok || len(configList) == 0 {
+		return nil
+	}
+	return configList[0].(map[string]interface{})
```

Comment:
this should be type asserted

Reply:
we already do type assertion at the beginning of this function, how exactly would it be different?

Reply:
Well, I can remove "ok" check, because it doensn't matter. But I can't cast it in one line, golang doesn't let it do so. That's why there is two steps.

Reply:
after discussing internally, this has been addressed